vumi_http_proxy/http_proxy.py

```diff
@@ -23,7 +25,21 @@ def buildProtocol(self, addr):
 class CheckProxyRequest(proxy.ProxyRequest):
     def process(self):
         host, _, port = self.getAllHeaders()['host'].partition(':')
-        if host in self.channel.blacklist:
+        self.d = client.getHostByName(host)
+        self.d.addCallback(self.setIP, host)
+        self.d.addErrback(printError)
```

Comment:
You can use `twisted.python.log` to print errors:
```python
from twisted.python import log
...
d.addErrback(log.err)
```
You probably also don't need to keep `d` on `self`, but you might want to return `d` from `process` (I'm not 100% sure where the thing calling `process` expects a deferred, but you can check if `proxy.ProxyRequest.process` returns one by looking at its source code).